lightning/src/ln/onchaintx.rs

```diff
@@ -856,17 +869,43 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 		}
 	}
 
+	pub(crate) fn transaction_unconfirmed<B: Deref, F: Deref, L: Deref>(
+		&mut self,
+		txid: &Txid,
+		broadcaster: B,
+		fee_estimator: F,
+		logger: L,
+	) where
+		B::Target: BroadcasterInterface,
+		F::Target: FeeEstimator,
+		L::Target: Logger,
+	{
+		let mut height = None;
+		for entry in self.onchain_events_awaiting_threshold_conf.iter() {
+			if entry.txid == *txid {
+				height = Some(entry.height);
+				break;
+			}
+		}
+
+		if let Some(height) = height {
+			self.block_disconnected(height, broadcaster, fee_estimator, logger);
```

Comment:
I think we could have `OnchainTxHandler::process_disconnect_event` call either by `transaction_unconfirmed` or `block_disconnected` sharing the API-independent code of current `block_disconnected`. By API-independent, I mean the regenerate-and-broadcast part which is not dependent on either txid or height.

I think it will make the code more proper rather than iterating twice on `onchain_events_awaiting_threshold_conf` and also make `OnchainTxHandler` API-ready if we expose it as its own abstract interface in the future. Though it's can be done in a follow-up and for now we can swallow the bullet as it's internal logic.

Reply:
I think I understand what you're suggesting. Just to clarify, are you saying that `process_disconnect_event` is the name of the proposed method refactored from `block_disconnected`?

Reply:
Yes `process_disconnect_event` would be the name of the block code extracted from `block_disconnected` !